File: mobile-app/lib/features/restaurant/domain/food_item_model.dart
import 'food_category_model.dart';

class FoodItem {
  final String id;
  final String name;
  final String description;
  final double price;
  final String? categoryId;
  final FoodCategory? category;
  final String image;
  final bool isAvailable;
  final int preparationTime;
  final List<String> tags;

  FoodItem({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    this.categoryId,
    this.category,
    required this.image,
    required this.isAvailable,
    required this.preparationTime,
    required this.tags,
  });

  factory FoodItem.fromJson(Map<String, dynamic> json) {
    return FoodItem(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      categoryId: json['category'] is String ? json['category'] : (json['category']?['_id']),
      category: json['category'] is Map<String, dynamic> ? FoodCategory.fromJson(json['category']) : null,
      image: json['image'] ?? 'https://images.unsplash.com/photo-1546793665-c74683f339c1?auto=format&fit=crop&q=80&w=800',
      isAvailable: json['isAvailable'] ?? true,
      preparationTime: json['preparationTime'] ?? 20,
      tags: List<String>.from(json['tags'] ?? []),
    );
  }
}

File: backend/src/models/foodItem.model.js
const mongoose = require('mongoose');

const foodItemSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: [true, 'Please add a food name'],
      trim: true,
    },
    description: {
      type: String,
      required: [true, 'Please add a description'],
      maxlength: [500, 'Description can not be more than 500 characters'],
    },
    price: {
      type: Number,
      required: [true, 'Please add a price'],
      min: [0, 'Price must be greater than or equal to 0'],
    },
    category: {
      type: mongoose.Schema.ObjectId,
      ref: 'FoodCategory',
      required: true,
    },
    image: {
      type: String,
      default: 'default-food.jpg',
    },
    imagePublicId: {
      type: String,
    },
    isAvailable: {
      type: Boolean,
      default: true,
    },
    preparationTime: {
      type: Number,
      default: 20,
    },
    tags: {
      type: [String],
    },
    isPopular: {
      type: Boolean,
      default: false,
    },
    features: {
      type: [String],
    },
    section: {
      type: String,
      enum: ['restaurant_bar', 'room_service'],
      default: 'restaurant_bar',
    },
  },
  {
    timestamps: true,
  }
);

foodItemSchema.index({ category: 1, isAvailable: 1 });
foodItemSchema.index({ name: 'text', description: 'text' });

module.exports = mongoose.model('FoodItem', foodItemSchema);
